pkg/kubelet/pod_workers.go

```diff
@@ -323,7 +324,7 @@ func killPodNow(podWorkers PodWorkers, recorder record.EventRecorder) eviction.K
 				PodStatusFunc: func(p *v1.Pod, podStatus *kubecontainer.PodStatus) v1.PodStatus {
 					return status
 				},
-				PodTerminationGracePeriodSecondsOverride: gracePeriodOverride,
+				PodTerminationGracePeriodSecondsOverride: utilptr.Int64Ptr(gracePeriod),
```

Comment:
The apiserver logic considers the spec value already if you pass nil here, i.e. omitempty removes it from the marshalled options.

Now you are basically replicating the logic that we have already. I would prefer to keep it only at one spot where we have it already according to API specification.

Reply:
The correct fix is to not pass a non-nil pointer to a zero value as override to this func.